Sage/carbon
# feat(dialog, dialog-full-screen, confirm): generate IDs dynamically (FE-4700)

The `Heading` component, used within `Dialog`, `DialogFullScreen` and `Confirm`, has its `titleId` and `subtitleId` props set to fixed values. This is problematic as these IDs aren't unique, potentially leading to multiple DOM elements having the same ID. 

### Proposed behaviour



- Replace fixed values with unique ids generated at runtime.
- Accessibility of the three components should not be impacted.

Example in `Dialog`:
![Screenshot 2022-01-17 at 11 33 37](https://user-images.githubusercontent.com/18368713/149762639-a4a73078-1a8c-4fcb-9aba-c9fb1a7c0e6f.png)

### Current behaviour



- `titleId` and `subtitleId` props of `Heading` are permanently set to `carbon-dialog-title` and `carbon-dialog-subtitle` respectively, potentially leading to a clash in ID values.

Example with `Dialog`:
![Screenshot 2022-01-17 at 11 28 34](https://user-images.githubusercontent.com/18368713/149762377-2e73fb96-9071-466c-99b5-d126d3a6effa.png)

### Checklist




#### QA


### Testing instructions

Open components in Storybook, check that no accessibility violations have been introduced.
